wlm/src/main/java/ai/djl/serving/wlm/Adapter.java

```diff
@@ -133,60 +130,40 @@ public String getSrc() {
     public <I, O> void register(WorkerPool<I, O> wp) {
         ModelInfo<I, O> wpc = (ModelInfo<I, O>) wp.getWpc();
         wpc.registerAdapter(this);
-        for (WorkerGroup<I, O> wg : wp.getWorkerGroups().values()) {
-            for (WorkerThread<I, O> t : wg.getWorkers()) {
-                t.addConfigJob(registerJob(wpc, t.getThreadType()));
-            }
-        }
+
+        // Add the register adapter job to job queue.
+        // Currently if enable_lora=true, we set tensor_parallel_degree=max,
+        // so that there will only be one worker
+        wp.getJobQueue().offer(registerJob(wpc));
     }
 
     /**
      * Creates a {@link WorkerJob} to register this adapter in a {@link WorkerThread}.
      *
      * @param wpc the worker pool of the thread
-     * @param threadConfig the thread config to register
      * @param <I> the input type
      * @param <O> the output type
      * @return the registration job
      */
-    public <I, O> WorkerJob<I, O> registerJob(
-            WorkerPoolConfig<I, O> wpc, ThreadConfig<I, O> threadConfig) {
-        ModelInfo<I, O>.ModelThread t = (ModelInfo<I, O>.ModelThread) threadConfig;
-        Job<I, O> job =
-                new Job<>(
-                        wpc,
-                        null,
-                        in -> {
-                            registerPredictor(t.getPredictor());
-                            return null;
-                        });
+    public <I, O> WorkerJob<I, O> registerJob(WorkerPoolConfig<I, O> wpc) {
+        Job<I, O> job = new Job<>(wpc, getRegisterInput());
         return new WorkerJob<>(job, new CompletableFuture<>());
     }
 
     /**
      * Creates a {@link WorkerJob} to unregister this adapter from a {@link WorkerThread}.
      *
      * @param wpc the worker pool of the thread
-     * @param threadConfig the thread config to unregister
      * @param <I> the input type
      * @param <O> the output type
      * @return the unregistration job
      */
-    public <I, O> WorkerJob<I, O> unregisterJob(
-            WorkerPoolConfig<I, O> wpc, ThreadConfig<I, O> threadConfig) {
-        ModelInfo<I, O>.ModelThread t = (ModelInfo<I, O>.ModelThread) threadConfig;
-        Job<I, O> job =
-                new Job<>(
-                        wpc,
-                        null,
-                        in -> {
-                            unregisterPredictor(t.getPredictor());
-                            return null;
-                        });
+    public <I, O> WorkerJob<I, O> unregisterJob(WorkerPoolConfig<I, O> wpc) {
+        Job<I, O> job = new Job<>(wpc, getUnregisterInput());
         return new WorkerJob<>(job, new CompletableFuture<>());
     }
 
-    protected abstract void registerPredictor(Predictor<?, ?> predictor);
+    protected abstract <I> I getRegisterInput();
```

Comment:
can we rename to `getRegisterAdapterInput`, and similarly for below `getUnregisterAdapterInput`

Reply:
Changed.